### ISSUE TITLE
Bitget error when use fetch_deposit_address

### ISSUE THREAD

**Issue by AlwxDavydov** [ccxt/ccxt#21112]
### Operating System

Windows 10 

### Programming Languages

Python

### CCXT Version

4.1.84

### Description

There is such a problem in bitget. I can't get the deposit address if there is a dollar sign in the ticker($AI, $ALT for example), I will get `{"code":"42017","msg":"The current chain does not support deposit the coin"` error. I removed the dollar using commonCurrency, but it doesn't help. I also noticed that commonCurrency does not work with bitget. For example, I included Toncoin there, but I will receive the following error - `code":"40019","msg":"Parameter coin cannot be empty`. I have not seen this with other exchanges. I also understand that if i put 'ALT':'ALT0', $ALT':'ALT', in commonCurrencies I always will got None as a result of my funcion but I dont know how to deal with it. 
  I also want to add, that after some test I think that this problem with obtaining a deposit address while currency in commonCurrencies occurs not only in bitget. For example, with gate and 'GTC'(Game.com in commonCurrencies) if we try exchange.fetchNetworkDepositAddress('GTC'), we will get  gate does not have currency code GTC, but if we try exchange.fetchNetworkDepositAddress('Game.com') we will get gate {"label":"INVALID_CURRENCY","message":"invalid currency"}. 
  With mexc for example if commonCurrency = {'GMT':'GOMINING, 'STEPN':'GMT',} if i use exchange.fetchDepositAddressesByNetwork('GMT') and exchange.fetchDepositAddressesByNetwork('GOMINING') I will get the same networks and addresses which are related only to STEPN GMT/ So if I use exchange.fetchDepositAddressesByNetwork('GOMINING') I will get wrong networks and sometimes addresses.
  I also admit that I did something wrong. In any case, it would be nice if you checked it out.

**Comment by ttodua** [comment]
thanks for report! 
I will check it out now.

**Comment by AlwxDavydov** [comment]
@ttodua Could you please tell me which exchanges this bug occurs with? I mentioned bitget, mexc and gate, but there are probably others

**Comment by ttodua** [comment]
@AlwxDavydov 
I have found out where the ccxt had issue with regard of those spec symbols, and pushed a fix.
thanks for your observation, i'll gradually fix on other exchanges, if you can, provide the names of exchanges where you faced that issue (metnion also method, `fetchDepositAddress` or whichever)
thanks

**Comment by AlwxDavydov** [comment]
@ttodua I'll write what I noticed. On the MEX exchange, when cyclically adding a currency to the CC, we combine two currencies into one and will only receive the one whose ticker is on the MEX

**Comment by AlwxDavydov** [comment]
@ttodua I'll write what I noticed. On the MEX exchange, when cyclically adding a currency to the CC, we combine two currencies into one and will only receive the one whose ticker is on the MEX

**Comment by AlwxDavydov** [comment]
@ttodua I'll write what I noticed.
  On the MEXC exchange, when cyclically adding a currency to the commonCurrencies, we combine two currencies into one and will only receive the one whose ticker is on the MEXC. It can be difficult to understand. Here's an example. commonCurrencies = {'GMT':'GOMINING','STEPN':'GMT',}. And if we try fetchDepositAddressesByNetwork we will get two identical options in any currency. And the information will relate to the ticker STEPN all the time and ignore GOMINING at all. This behavior can be misleading and cause errors. 
  With the Kucoin exchange. If the token is in commonCurrencies. The only way to get data, to use fetch_deposit_address, and fetchDepositAddressesByNetwork will not work for thease tokens.
  With Gate and Bitget, it is just impossible to fetch deposit data while token in commonCurrencies. 
  I would also like to note that on Binance, Bybit and Huobi everything works as expected. So you can do  commonCurrencies = {'TWT':'BTC','BTC':'TWT', And when you do fetchDepositAddressesByNetwork('BTC') you will get data about TWT and vice versa. So Id like to see same behavior with gate, kucoin and bitget. I also want to mencion that I only tried deposit functions and do not know will the bug work with withdrawal or other methods.

**Comment by AlwxDavydov** [comment]
@ttodua Hello, is there any news regarding the problem with the commonCurrencies. Also note that I checked and all the above problems with commonCurrencies and mexc\gate\bitget also apply to fetchtikers, fetch_deposit_withdraw_fee and others. Maybe I do something wrong, please tell me.

**Comment by ttodua** [comment]
@AlwxDavydov 
No alex, you are doing correctly, there is issue with special-symbol encoding and I'll be fixing them, sorry for delay because I god sidetracked with other tasks, and will soon address these problems, within next 1-2 days

**Comment by AlwxDavydov** [comment]
@ttodua Please tag me in this ticket when the issue will be resolved. I will help you to test all of that.

**Comment by AlwxDavydov** [comment]
@ttodua Hello, are there any news about this ticket. I recently tried to do the same thing, but the problem apparently is not solved

**Comment by carlosmiei** [comment]
@AlwxDavydov Can you post a verbose log where we can see the request+response?

Enable verbose mode by doing this:
```
await exchange.load_markets()
exchange.verbose = True
```

**Comment by ttodua** [comment]
@AlwxDavydov do you mean bitget exchange again? or other exchanges you mentioned in recent comments.
I'll be checking the things tomorrow and let you know

**Comment by AlwxDavydov** [comment]
@ttodua I mean other exchanges mentioned. I described in detail all the problems with every exchange that I faced in the comment on February 8. I also ran a query for characters that have a dollar sign on bitget and they work well now, thank you. I just want you to test all described errors, maybe some of them are fixed now. But I tried some of them yesterday and face same problems again

**Comment by AlwxDavydov** [comment]
@ttodua @carlosmiei I updated ccxt and checked everything again. It looks like there is no longer a problem with Mexc, but bitget and gate still work incorrectly. You simply cant use fetch_deposit_address, fetch_deposit_withdraw_fee etc with all currencies added to common currency.

**Comment by AlwxDavydov** [comment]
and also with kucoin If the token is in commonCurrencies. The only way to get data, to use fetch_deposit_address, and fetchDepositAddressesByNetwork will not work for thease tokens.

**Comment by ttodua** [comment]
sorry, i got sidetracked. I'll look into that today

**Comment by ttodua** [comment]
@AlwxDavydov can you tell, which coin didnt work with bitget? i see bitget was fixed and I can successfully i.e. `fetchDepositAddress ('$AI')`

**Comment by ttodua** [comment]
also, please tell which coin didnt work with mexc. I cant find coins which include `$` character in mexc.

**Comment by AlwxDavydov** [comment]
@ttodua Yes, that's how it will work. But in the process of studying this problem, I came across a new one, with commonCurrencies. Which I described in detail above. In short, with a bitget you will not be able to interact with the coin that is in commonCurrencies, you cant use fetch_deposit_address, fetch_deposit_withdraw_fee and some other with all currencies added to common currency.

**Comment by AlwxDavydov** [comment]
If you want I can open a new issue and describe it there

**Comment by ttodua** [comment]
ok, thanks for reply. so, we should divide the subject into two problems:
1) coins which include `$` (or alike) character and doesnt work on exchanges (please list such exchanges if you know)
2) the "commonCurrencies" issue which is independent `$` character

**Comment by AlwxDavydov** [comment]
Yes, correct

**Comment by AlwxDavydov** [comment]
As far as I know there is a problem with tokens which include $ in solved now. At least on Bitget everything works well, but I haven’t seen this on other exchanges. And the problem with the commonCurrencies has not been described before, as far as I know. And it quite seriously affects the functions of some exchanges and could be misleading.

**Comment by ttodua** [comment]
can you tell me what do you get when using `mexc fetchDepositAddress GMT` (verbose output, sensitive info removed).
I am successfully getting `STEPN` coin, which is mapped in commonCurrencies, and that is expected behavior

**Comment by ttodua** [comment]
specifically for mexc, at first, you have to use `.createDepositAddress()` to generate a deposit address, after that, you can `fetchDepositAddress`

**Comment by ttodua** [comment]
I have also tried with `GMT Token` which is in commonCurrencies:  `await ex.fetchDepositAddress ('GMT Token', {network: 'ERC20'});` and it successfully fetches `GMT Token`, which is also expected.

**Comment by ttodua** [comment]
with kucoin, fetchDepositAddress seems not to work for any other currencies i.e. `SHIB` or etc, so it seems to have broken implementation.

**Comment by AlwxDavydov** [comment]
@ttodua As I said yesterday(https://github.com/ccxt/ccxt/issues/21112#issuecomment-2027537407) with Mexс this problem with commonCurrencies no longer exists. However it stays with gate, bitget and kucoin. With Gate and Bitget, it is just impossible to fetch deposit data while token in commonCurrencies.

**Comment by ttodua** [comment]
but i've tested myself with bitget using currencies that are in commonCurrencies and it worked for me. that's why I asked to provide exact command (code line, i.e. `ex.fetchXyz ('Xyz', params..)` ) you are running, while having verbose output enable at this moment, and post it here

**Comment by AlwxDavydov** [comment]
Okay, with bitget: 'commonCurrencies': {'TONCOIN':'TON'}, use metods .fetch_deposit_address('TON') and I will recieve ccxt.base.errors.BadRequest: bitget {"code":"40019","msg":"Parameter coin cannot be empty","requestTime":1711832366119,"data":null}. Or use .fetch_deposit_withdraw_fee('TON') and I will recieve None.

**Comment by ttodua** [comment]
> Okay, example with bitget: 'commonCurrencies': {'TONCOIN':'TON'}, use metods .fetch_deposit_address('TON') and I will recieve ccxt.base.errors.BadRequest: bitget {"code":"40019","msg":"Parameter coin cannot be empty","requestTime":1711832366119,"data":null}. Or use .fetch_deposit_withdraw_fee('TON') and I will recieve None.

thanks for follow up, i've successfully reproduced the issue and confirm it. Working on a fix and will be available in few hours

**Comment by AlwxDavydov** [comment]
@ttodua I also want to repeat that there is a similar problem with the gate. 'commonCurrencies': {'NANO':'XNO'}, .fetchDepositAddressesByNetwork('XNO'), I get ccxt.base.errors.BadSymbol: gate {"label":"INVALID_CURRENCY","message":"invalid currency"}, if I use .fetchDepositAddressesByNetwork('NANO'), I get ccxt.base.errors.ExchangeError: gate does not have currency code NANO, same with withdrawl functions. And aslo Kucoin have same problem, but if I am not mistaken you already found it

**Comment by ttodua** [comment]
> @ttodua I also want to repeat that there is a similar problem with the gate. 'commonCurrencies': {'NANO':'XNO'}, .fetchNetworkDepositAddress('XNO'), I get ccxt.base.errors.BadSymbol: gate {"label":"INVALID_CURRENCY","message":"invalid currency"}, if I use .fetchDepositAddressesByNetwork('NANO'), I get ccxt.base.errors.ExchangeError: gate does not have currency code NANO, same with withdrawl functions. And aslo Kucoin have same problem, but if I am not mistaken you already found it. But I can show it again. With kucoin 'commonCurrencies': {'NANO':'XNO'}, .fetchDepositAddressesByNetwork('XNO'), I will get {}

yes, thanks again, I confirm what you have been described was true and there was a bug. the bug is a kind of random nature, which means it depends on exchanges and i have to check exchanges manually, some of them might have that bug, some not. bitget is already fixed (PR will be merged soon) and then i'll check gate, kucoin and other toptier exchanges manually, and will link the update PRs to this issue, so you will also get notifications. note that, after PR is merged, it will need few hours before it's published and you can update your package.  let me know all exchanges where you might face any issues.

**Comment by AlwxDavydov** [comment]
@ttodua Also pay attention to mexc. I said that it was fixed but it is strange case. For me, it literally doesn’t always produce what I need and I don’t know what it depends on. Try 'commonCurrencies': {'BANANAGUN':'BANANA',} and .fetch_deposit_withdraw_fee, it will probably return None but sometimes it will return the right answer.

**Comment by ttodua** [comment]
ok, i'll look into mexc too

**Comment by ttodua** [comment]
> fetch_deposit_withdraw_fee

Alex, I've tried with that `BANANAGUN:BANANA`, but I do get answer. If you have some deterministic way to reproduce error, please let me know, because I can't reproduce the issue with mexc by your explanation. 
However, I have a minor note: by that specific example I can say that it might have some glitches because when you are using `X: Y` in common currencies, the second side `Y` should be a code that does not exist in currencies. However, MEXC actually has tokens: `BANANA, BANANAOLD, BANANAGUN` and as you used `BANANA` , MEXC already contains `BANANA` code, and it might conflict. so, I should advise at first to remap the existing `BANANA` and then remap `BANANAGUN` to `BANANA`:
```
commonCurrencies : {
    ....
    'BANANA': 'SOMETHING_OTHER_YOU_WANT',
    'BANANAGUN': 'BANANA',
    ....
}
```
and let me know

**Comment by AlwxDavydov** [comment]
@ttodua Yes you are right. And you helped me find how to reproduce the error. Try 'commonCurrencies': {'BANANA':'NOTBANANA', 'BANANAGUN':'BANANA',} and .fetch_deposit_withdraw_fee('BANANA') will return None, and .fetch_deposit_withdraw_fee('NOTBANANA') will return the right answer for token with Ticker 'Banana' on mexc.

**Comment by ttodua** [comment]
good! i've now reproduced that issue. working on it.

**Comment by ttodua** [comment]
PR is underway, there were two separate issues in the lib:  
- `fetch_deposit_withdraw_fee` - ignored commonCurrencies values (fixed)
- commonCurrencies parsing issue with currency that was defined as key (i.e. BANANA) already and same string (`BANANA`) assigned to other member (fixed)

somewhile after that PR is merged, update ccxt to have a fixed version

**Comment by ttodua** [comment]
@AlwxDavydov the changes had been merged and when you'll see https://pypi.org/project/ccxt/ 4.89 version released, test it out and please repost if there are still any exchanges where there is a bug

**Comment by AlwxDavydov** [comment]
@ttodua  I installed ccxt version 4.2.89 and checked the errors. Here's what I noticed, each item is a separate error and an explanation of how to get it.

1) On kucoin fetchDepositAddressesByNetwork (probably some other functions too) will not work if token in commonCurrencies. Try 'commonCurrencies':{'KALT':'ALT'} and .fetchDepositAddressesByNetwork(’ALT’) it will return {}

2) On gate fetchDepositAddressesByNetwork will not work if token in commonCurrencies too. Try 'commonCurrencies':{'NANO':'XNO',} and .fetchDepositAddressesByNetwork(’XNO’) it will return *BadSymbol: gate {"label":"INVALID_CURRENCY","message":"invalid currency"}*

3) On mexc, if you remap a specific currency it will not give effect and return the wrong currency. Try 

```python
'commonCurrencies' :{
  'BANANA': 'BANANA_BSC',
  'BANANAGUN': 'BANANA'}
```

and fetch_deposit_withdraw_fee(’BANANA’), in fact it will return the token named BANANA_BSC for me, you can confirm it if you dive into data, in ‘networks’ you can find data about BSC, and BANANAGUN have no BSC contract, so I can assume that token BANANA was not remaped at all, so I get wrong data. You can also try it with {'ALT':'APTOS_ALT', 'ALTLAYER':'ALT',}  and you will get same result.

**Comment by ttodua** [comment]
thanks, i'll take a look in the upcoming morning

**Comment by AlwxDavydov** [comment]
@ttodua I also notice one more thing. All thing with bitget work well now, except cycle remapping. if you remap a specific currency it will not give effect and return the wrong currency. Try

```python
'commonCurrencies' :{
  'ALT':'ALT_ARCH',
  '$ALT':'ALT',
  }
```

and fetch_ticker('ALT/USDT'), in fact it will return ticker for ALT_ARCH. But this bug did not affect fetch_deposit_withdraw_fee and other withdraw\deposit functions, what's most interesting.

**Comment by sc0Vu** [comment]
@carlosmiei @ttodua  I think we should think how to use common currencies thoroughly. To implement cyclically behavior or not?
BTW, it used to be simple, just replacement.

**Comment by ttodua** [comment]
@sc0Vu i think that PR would fix all these things. there, as seen, i've reverted commonCurrencyCode method, so it is simple replacement as you say. the problem, was that, as discussed in discord thread, the `fetchDepositWithdrawFees` was the only method which had bug in base, and that is also solved. I think, after merging our combined PRs, all the problems will be solved

**Comment by ttodua** [comment]
so, what @AlwxDavydov mentions in last above comments, should also be fixed across all exchanges (unless some specific exchange has incorrect implementation. what I have seen, such buggy exchange was only bitget and it was also fixed). mexc, gate, kucoin etc.. seem to have correct implementations and they will be automatically fixed with that pending PR

**Comment by AlwxDavydov** [comment]
@ttodua No problem. I will help you to test new version

**Comment by ttodua** [comment]
fixed is on the way, in around 5-8 hours, once new CCXT package released, you can update and retest.  mexc, gate, bitget... should be fixed with all methods you can interact

**Comment by ttodua** [comment]
@AlwxDavydov new package published. can you please update now and retest and let me know if you face any issues? thanks

**Comment by AlwxDavydov** [comment]
@ttodua I installed ccxt version **4.2.91. I tested almost everything and could only find two errors. It also was in previous version.**

1. On kucoin fetchDepositAddressesByNetwork will not work if token in commonCurrencies. Try 'commonCurrencies':{'KALT':'ALT'} and .fetchDepositAddressesByNetwork(’ALT’) it will return {}
2. On gate fetchNetworkDepositAddress will not work if token in commonCurrencies too. Try 'commonCurrencies':{'NANO':'XNO',} and .fetchNetworkDepositAddress (’XNO’) it will return *BadSymbol: gate {"label":"INVALID_CURRENCY","message":"invalid currency"}*

Bitget, mexc(cyclic reassignment including) and other big exchanges work well now.

**Comment by ttodua** [comment]
@AlwxDavydov thanks! 
I'll get to those remnant errors, they are probably specific implementation errors in gate exchange and kucoin

**Comment by ttodua** [comment]
@AlwxDavydov I have checked kucoin and I think the problem sees expected, that's because you have only `{'KALT':'ALT'}` mapping, but kucoin seems to have `ALT` coin already. so you need to add it's mapping (to other name) so your conflicting ids would work
```
{
   'ALT':' OTHER_ALT',
   'KALT': 'ALT',
}
```
I think same problem might be with gate, please try and let me know

**Comment by ttodua** [comment]
one more note i see now, kucoin exchange specifically might need to have such logic "if fetchDepositAddress result is empty, then try createDepositAddress at first, and then re-try fDA".

**Comment by ttodua** [comment]
i tested myself and confirmed what I said above. so, i'll close this issue as completed (revise it and let me know if that doesnt work for you).
but with gate, I see that issue is very specific to gate and I've opened a new issue https://github.com/ccxt/ccxt/issues/22117 (and let's migrate there)

**Comment by AlwxDavydov** [comment]
hello @ttodua please forgive me for the long response, I was busy. First about kucoin. I tried to do what you say, but I still didn't get the expected result. Perhaps I'm doing something wrong.

```python
{
   'ALT':' APTOS_ALT',
   'KALT': 'ALT',
}
.fetchDepositAddressesByNetwork('ALT')
.fetchDepositAddressesByNetwork('APTOS_ALT')
```

In both cases I get {}

I also know that I have to create deposit address by myself but I did it already and if I try to do it again I will get *ccxt.base.errors.InvalidAddress: kucoin Deposit address already exists.*

Moreover I tried this

```python
{
   'QI': 'BENQI',
}
.fetchDepositAddressesByNetwork('BENQI')
.fetchDepositAddressesByNetwork('QI') #(just to make sure)
```

And again I will get {}, I created address already too. I am sure about that. And kucoin does not have ticker BENQI. So I don t know what is the problem.

**Comment by ttodua** [comment]
@AlwxDavydov sorry for delayed response. I had to ask, did you set `commonCurrencies` before markets are loaded? it should be done so, because otherwise it will not work as expected.

**Comment by ttodua** [comment]
wait, i see the issue again. give me few minutes

**Comment by AlwxDavydov** [comment]
@ttodua Okay, got you. Please clarify should I open a new issue? Because I know that there is already a issue with such a problem. (https://github.com/ccxt/ccxt/issues/22117)

**Comment by ttodua** [comment]
> @ttodua Okay, got you. Please clarify should I open a new issue with gate? Because I know that there is already a issue with such a problem. (#22117)

ahh, right, seems I have opened an issue and assigned myself. ok, you can comment/subscribe in that issue whenever you want or ping me there, if i will not have an update there in upcoming period.

**Comment by AlwxDavydov** [comment]
@ttodua I subscribed for the gate problem. Please tell me if I should write again what exactly the problem is. Also, please tell me which version of the ccxt will contain the kucoin fix, so that I can track its release on pypi

**Comment by ttodua** [comment]
kucoin was merged successfully a few hours ago ( you can take a look into this too: https://app.travis-ci.com/github/ccxt/ccxt/builds ), you can update ccxt to use fixed version for kucoin and let me know if the above problem for kucoin is fixed

**Comment by AlwxDavydov** [comment]
@ttodua Hello, sorry for delay. I checked kucoin. So, in version 4.3.11 If I define a currency in commonCurrencies, then the methods will also not work with this. Try ccxt.kucoin().fetch_ticker('ALT') and you will get kucoin does not have market symbol ALT. Alt is a basic defined token in CC, so you do not need add it here. I also still recieve {} when try to fetchDepositAddressesByNetwork. Try it please.

**Comment by ttodua** [comment]
Thanks, i'll recheck

**Comment by ttodua** [comment]
@AlwxDavydov `fetchTicker()` will not work with plain coin names (i.e. `ALT`) instead you need to provide pair (not coin) i.e. `ALT/USDT`

**Comment by ttodua** [comment]
I also tried now `.fetchDepositAddressesByNetwork('ALT') ` and I do get address successfully. did you update ccxt to latest version? did you user `.createDepositAddress('ALT')` at first?

**Comment by AlwxDavydov** [comment]
@ttodua Oh my god, yes of course you are right. I've been busy lately, but I really wanted to answer you, in the end I did it being tired, making a stupid mistake. I'm sorry. Now I have rechecked everything and it really works as it should. However, I still ran into an issue with fetch_deposit_addresses_by_network, however it was caused by the kucoin api settings. In the end, everything works as it should. It seems that only the gate left. Thank you!
